src/console.cpp

```diff
@@ -167,6 +168,27 @@ void	toggleConsoleDrop()
 	}
 }
 
+bool addConsoleMessageDebounced(const char * Text, CONSOLE_TEXT_JUSTIFICATION jusType, SDWORD player, const DEBOUNCED_MESSAGE & message, bool team)
+{
+	// Messages are debounced individually - one debounced message won't stop a different one from appearing
+	static std::map<const DEBOUNCED_MESSAGE *, std::chrono::milliseconds> nextAllowedMessageTimes;
+
+	const std::chrono::milliseconds DEBOUNCE_TIME(message.debounceTime);
+	const std::chrono::milliseconds now = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now().time_since_epoch());
+
+	const auto nextAllowedMessageTime = nextAllowedMessageTimes.find(&message);
+
+	if (nextAllowedMessageTime == nextAllowedMessageTimes.end() || now > nextAllowedMessageTime->second)
+	{
+		nextAllowedMessageTimes.insert_or_assign(&message, now + DEBOUNCE_TIME);
```

Comment:
Error from CI:
> ../src/console.cpp:183:27: error: 'class std::map<const DEBOUNCED_MESSAGE*, std::chrono::duration<long int, std::ratio<1, 1000> > >' has no member named 'insert_or_assign'
  183 |   nextAllowedMessageTimes.insert_or_assign(&message, now + DEBOUNCE_TIME);

Alas, at the moment we're restricted to C++11.

Reply:
Ah, thanks for letting me know, I'll dial it down a bit.  It's an over-engineered solution for the reported issue anyway, and I don't want to over-step while I'm still saying Hello, Warzone.  I might dial it back to putting the debouncing just in the "Can't build" function.

CMakeLists.txt does actually specify a different CXX for MSVC - I'm not sure why, or if that's deliberate.  It rather seems to be inviting issues like this.

if(MSVC)
	set(CMAKE_CXX_STANDARD 14)
else()
	set(CMAKE_CXX_STANDARD 11)
endif()